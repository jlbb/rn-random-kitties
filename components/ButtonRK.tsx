import { StyleSheet, Text, View, TouchableHighlight } from "react-native";
import React from "react";
import { debounce } from "lodash";
import { colors } from "../config/contants";

interface ButtonRKProps {
  title: string;
  type?: ButtonRKTypes;
  onButtonPress: () => any;
}

const ButtonRK = ({
  title,
  type = DEFAULT,
  onButtonPress
}: ButtonRKProps): JSX.Element => {
  const buttonStyle: any =
    type === PRIMARY
      ? StyleSheet.flatten([styles.primaryButton])
      : StyleSheet.flatten([styles.defaultButton]);

  const _onButtonPress: any = debounce(() => {
    onButtonPress();
  }, 300);

  return (
    <View style={styles.container}>
      <TouchableHighlight
        style={[
          styles.button,
          { backgroundColor: buttonStyle.backgroundColor }
        ]}
        onPress={_onButtonPress}
      >
        <Text
          style={[
            styles.content,
            {
              color: buttonStyle.color,
              fontSize: buttonStyle.fontSize
            }
          ]}
        >
          {title}
        </Text>
      </TouchableHighlight>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  button: {
    minHeight: 50,
    height: "100%",
    flexDirection: "row",
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: colors.DEFAULT_BUTTON_UNDERLAY_DEFAULT
  },
  content: {
    fontSize: 18,
    color: colors.DEFAULT_BUTTON_TEXT_DEFAULT,
    textTransform: "uppercase"
  },
  defaultButton: {
    backgroundColor: colors.DEFAULT_BUTTON_UNDERLAY_DEFAULT,
    color: colors.DEFAULT_BUTTON_TEXT_DEFAULT,
    fontSize: 16
  },
  primaryButton: {
    backgroundColor: colors.PRIMARY_BUTTON_UNDERLAY_DEFAULT,
    color: colors.DEFAULT_BUTTON_TEXT_DEFAULT,
    fontSize: 16
  }
});

export const DEFAULT = "DEFAULT";
export const PRIMARY = "PRIMARY";

export type ButtonRKTypes = typeof DEFAULT | typeof PRIMARY;

export default ButtonRK;
