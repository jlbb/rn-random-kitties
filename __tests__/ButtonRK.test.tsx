import ButtonRK, { PRIMARY } from "../components/ButtonRK";
import { mount, shallow } from "enzyme";

describe("ButtonRK", () => {
  const buttonAction = jest.fn();

  it("renders correctly", () => {
    expect(
      shallow(<ButtonRK title={"Click me!"} onButtonPress={buttonAction} />)
    ).toMatchSnapshot();
  });

  it("executes a callback when is clicked", () => {
    const wrapper = mount(
      <ButtonRK title={"Click me"} onButtonPress={buttonAction} />
    );

    wrapper.prop("onButtonPress")();

    expect(buttonAction).toHaveBeenCalled();
  });

  it("renders with PRIMARY type style", () => {
    const wrapper = shallow(
      <ButtonRK
        title={"Click me"}
        onButtonPress={buttonAction}
        type={PRIMARY}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
