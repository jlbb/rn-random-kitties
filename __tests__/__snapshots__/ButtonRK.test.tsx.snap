// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonRK renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#984fa0",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "minHeight": 50,
        },
        Object {
          "backgroundColor": "#984fa0",
        },
      ]
    }
    underlayColor="black"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#f5f5f5",
            "fontSize": 18,
            "textTransform": "uppercase",
          },
          Object {
            "color": "#f5f5f5",
            "fontSize": 16,
          },
        ]
      }
    >
      Click me!
    </Text>
  </TouchableHighlight>
</View>
`;

exports[`ButtonRK renders with PRIMARY type style 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#984fa0",
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "center",
          "minHeight": 50,
        },
        Object {
          "backgroundColor": "#229a80",
        },
      ]
    }
    underlayColor="black"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#f5f5f5",
            "fontSize": 18,
            "textTransform": "uppercase",
          },
          Object {
            "color": "#f5f5f5",
            "fontSize": 16,
          },
        ]
      }
    >
      Click me
    </Text>
  </TouchableHighlight>
</View>
`;
